misc/python/materialize/mzcompose/__init__.py

```diff
@@ -452,6 +452,11 @@ def override(self, *services: "Service") -> Iterator[None]:
 
         self._write_compose()
 
+        for service in services:
+            # Perform 'docker pull' on any 'latest' images in order to ensure freshness
+            if "image" in service.config and "latest" in service.config["image"]:
```

Comment:
`"latest" in service.config["image"]` is a bit too imprecise. If we want to limit to `latest` then we need to split on `:` and check that the part after the colon is `latest`. But I think special-casing `latest` is probably too specific, and we want to unconditionally pull _all_ non-mzbuild images. There are plenty of other image tags that get updated periodically, like `ubuntu:rolling`.

Reply:
I am afraid pulling all non-mzbuild images causes noticable delay on test startup as "kafka", "schema-registry" , zookeeper and the like are pulled. So I have preserved the special case but extended it to "unstable" and "rolling" as well.